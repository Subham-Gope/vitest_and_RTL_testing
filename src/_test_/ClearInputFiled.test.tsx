// import { describe, test, expect } from "vitest";
// import { screen, render } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import ClearInputFiled from "../components/ClearInputFiled";

// describe("Testing the ClearInputFiled.tsx file", () => {
//   test("enter a text in the input and then clears it", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     const { container } = render(<ClearInputFiled />);
//     expect(container).toMatchSnapshot();

//     const input = screen.getByRole("textbox", { name: /^enter your text$/i });
//     expect(input).toBeInTheDocument();

//     // intiating user interactions
//     await user.type(input, "hello");
//     expect(input).toHaveValue("hello");

//     await user.clear(input);
//     expect(input).toHaveValue("");
//   });
// });
// //
