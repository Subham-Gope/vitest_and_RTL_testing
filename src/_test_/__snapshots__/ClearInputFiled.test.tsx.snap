// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the ClearInputFiled.tsx file > enter a text in the input and then clears it 1`] = `
<div>
  <div>
    <label
      for="clear"
    >
      Enter Your Text
    </label>
    <input
      id="clear"
      name="clear"
      placeholder="please enter your text"
      type="text"
      value=""
    />
  </div>
</div>
`;
